src/cmd/ksh93/include/nval.h

```diff
@@ -201,37 +227,45 @@ static inline void nv_setattr(Namval_t *np, unsigned int nvflag) { np->nvflag =
 
 static inline bool nv_isarray(Namval_t *np) { return nv_isattr(np, NV_ARRAY) == NV_ARRAY; }
 
-// The following are operations for associative arrays.
-#define NV_AINIT 1     // initialize
-#define NV_AFREE 2     // free array
-#define NV_ANEXT 3     // advance to next subscript
-#define NV_ANAME 4     // return subscript name
-#define NV_ADELETE 5   // delete current subscript
-#define NV_AADD 6      // add subscript if not found
-#define NV_ACURRENT 7  // return current subscript Namval_t*
-#define NV_ASETSUB 8   // set current subscript
-
-// The following are for nv_disc.
-#define NV_FIRST 1
-#define NV_LAST 2
-#define NV_POP 3
-#define NV_CLONE 4
+// The following symbols are for use with nv_disc(). We start with the arbitrary value 113 to help
+// ensure that calling `nv_disc()` with an unexpected op value (especially zero) will fail.
+enum {
+    DISC_OP_NOOP_val = 1,  // ??? (this used to be the magic `0` constant used by four callers)
```

Comment:
Yes and no. Yes, it was 113 before I switch from simple enums to structs. No, because with the typedef'd struct approach there isn't any compelling reason to avoid overlapping ranges. I'll fix the comment.